modular_skyrat/modules/interaction_menu/code/interaction_component.dm

```diff
@@ -113,4 +135,89 @@
+				source.visible_message(span_purple("[source.name] starts trying to [insert_or_attach] the [new_item.name] [into_or_onto] [target.name]'s [item_index]."), span_purple("You start to [insert_or_attach] the [new_item.name] [into_or_onto] [target.name]'s [item_index]."), span_purple("You hear someone trying to [insert_or_attach] something [into_or_onto] someone nearby."), vision_distance = 1, ignored_mobs = list(target))
+			if (source != target)
+				target.show_message(span_warning("[source.name] is trying to [existing_item ? "remove the [existing_item.name] [internal ? "in" : "on"]" : new_item ? "is trying to [insert_or_attach] the [new_item.name] [into_or_onto]" : span_alert("What the fuck, impossible condition? interaction_component.dm!")] your [item_index]!"))
+			if(do_after(
+				source,
+				5 SECONDS,
+				target,
+				interaction_key = "interation_[item_index]"
+				) && can_lewd_strip(source, target, item_index))
+				if(existing_item)
+					source.visible_message(span_purple("[source.name] removes [existing_item.name] from [target.name]'s [item_index]."), span_purple("You remove [existing_item.name] from [target.name]'s [item_index]."), span_purple("You hear someone remove something from someone nearby."), vision_distance = 1)
+					target.dropItemToGround(existing_item, force = TRUE) // Force is true, cause nodrop shouldn't affect lewd items.
+					target.vars[item_index] = null
+					target.update_inv_vagina()
+					target.update_inv_penis()
+					target.update_inv_anus()
+					target.update_inv_nipples()
+				else if (new_item)
+					source.visible_message(span_purple("[source.name] [internal ? "inserts" : "attaches"] the [new_item.name] [into_or_onto] [target.name]'s [item_index]."), span_purple("You [insert_or_attach] the [new_item.name] [into_or_onto] [target.name]'s [item_index]."), span_purple("You hear someone [insert_or_attach] something [into_or_onto] someone nearby."), vision_distance = 1)
+					target.vars[item_index] = new_item
+					new_item.forceMove(target)
+					target.update_inv_vagina()
+					target.update_inv_penis()
+					target.update_inv_anus()
+					target.update_inv_nipples()
+
+		else
+			source.show_message(span_warning("Failed to adjust [target.name]'s toys!"))
+
+		return TRUE
+
 	message_admins("Unhandled interaction '[params["interaction"]]'. Inform coders.")
+
+/// Checks if the target has ERP toys enabled, and can be logially reached by the user.
+/datum/component/interactable/proc/can_lewd_strip(mob/living/carbon/human/user, mob/living/carbon/human/other_user, slot_index)
+	if(!other_user.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
+		return FALSE
+	if(!(user.loc == other_user.loc || user.Adjacent(other_user)))
+		return FALSE
+	if(!user.has_arms())
+		return FALSE
+	if(!slot_index) // This condition is for the UI to decide if the button is shown at all. Slot index should never be null otherwise.
+		return TRUE
+	if(slot_index == "slot_nipples" && !other_user.is_topless())
```

Comment:
The define is different from what is sent back by TGUI... which I completely forgot to update to be the same as the defines, so thanks for reminding me!

Reply:
Haha, turns out the opposite is true, I did a dumb and forgot to update this to the right define, so that was dead code until the latest commit :pain: